gcs-fetcher/pkg/fetcher/fetcher.go

```diff
@@ -317,7 +354,11 @@ func (gf *Fetcher) fetchObjectOnce(ctx context.Context, j job, dest string, brea
 		result.err = fmt.Errorf("creating destination file %q: %v", dest, err)
 		return result
 	}
-	defer f.Close()
+	defer func() {
+		if err := f.Close(); err != nil {
+			log.Fatalf("Failed to close file %q: %v", dest, err)
```

Comment:
Same with `log.Fatalf`s, would probably be better to have these errors returned by the method and handled in `main.go` instead.

Reply:
This too seems to be pretty tough in Go. Can you provide an example? Simply "return err" doesn't work well from a defer (that I can see).

Reply:
Yeah `defer` definitely makes this harder, unfortunately :disappointed: 

It's sometimes easier if you use a named return value:

```go
func foo(wc io.WriteCloser) (err error) {
  defer func() {
    cerr := wc.Close()
    if err == nil { err = cerr }
  }()
  if err := w.Write(...); err != nil {
    return err
  }
  return nil
}
```

https://play.golang.org/p/wAoWS8g_KbM